CardGame/src/components/Card/CardGameContainer.tsx

```diff
@@ -0,0 +1,102 @@
+import { useState, useEffect } from "react";
+import { styled } from "styled-components";
+import { imageData } from "@/assets/data/imageData";
+import Card from "@/components/Card/Card";
+import BasicImg from "@/assets/images/basicMaru.webp";
+
+interface CardgameContainerInterface {
+  numPairs: number;
+  setScore: React.Dispatch<React.SetStateAction<number>>;
+  reset: boolean;
+}
+
+interface CardType {
+  id: number;
+  imgSrc: string;
+  pairId: number;
+}
+
+const CardgameContainer = ({
+  numPairs,
+  setScore,
+  reset,
+}: CardgameContainerInterface) => {
+  const [openedCards, setOpenedCards] = useState<number[]>([]);
+  const [matchedCards, setMatchedCards] = useState<number[]>([]);
+  const [cards, setCards] = useState<CardType[]>([]);
+  const [initialFlip, setInitialFlip] = useState(false);
+
+  useEffect(() => {
+    initCards();
+    setInitialFlip(true);
+    setTimeout(() => setInitialFlip(false), 200);
+  }, [numPairs, reset]);
+
+  const initCards = () => {
+    const selectedImages = imageData
+      .sort(() => 0.5 - Math.random())
+      .slice(0, numPairs);
+
+    const shuffledCards = selectedImages
+      .flatMap((image) => [
+        { id: image.id * 2, imgSrc: image.imgSrc, pairId: image.id },
+        { id: image.id * 2 + 1, imgSrc: image.imgSrc, pairId: image.id },
+      ])
+      .sort(() => Math.random() - 0.5);
+
+    setCards(shuffledCards);
+    setOpenedCards([]);
+    setMatchedCards([]);
+  };
+
+  const isFlipped = (cardId: number) => {
+    return openedCards.includes(cardId) || matchedCards.includes(cardId);
+  };
+
+  const onClickCard = (cardId: number) => {
+    if (isFlipped(cardId)) return;
```

Comment:
저는 &&연산자가 코드를 더 깔끔하게 보이게 하는 거 같아서 선호하는 편이랍니다 !
여기 부분 
```suggestion
    isFlipped(cardId) && return;
```
로 간단하게 줄여볼 수 있어요!